eslint-local-rules/disallowSpecImportSpec.js

```diff
@@ -0,0 +1,36 @@
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        // eslint-disable-next-line max-len
+        'Disallow importing spec file code inside a spec file, it would execute the tests from the imported spec file twice',
+      recommended: false,
+    },
+    schema: [],
+  },
+  create(context) {
+    if (!isSpecFile(context.getFilename())) {
+      return {}
+    }
```

Comment:
We could run this on all test files, to catch sub-imports (ex: foo.spec imports specHelper imports bar.spec)

Reply:
Let's check that everywhere then, an import on prod code would have the same behavior